agent/api/container.go

```diff
@@ -246,6 +248,33 @@ func (c *Container) GetKnownExitCode() *int {
 	return c.knownExitCode
 }
 
+// SetRegistryAuthCredentials sets the credentials for pulling image from ECR
+func (c *Container) SetRegistryAuthCredentials(credential credentials.IAMRoleCredentials) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.RegistryAuthentication.ECRAuthData.SetPullCredentials(credential)
+}
+
+// ShouldPullWithExecutionRole returns whether this container has its own ECR credentials
+func (c *Container) ShouldPullWithExecutionRole() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.RegistryAuthentication != nil &&
+		c.RegistryAuthentication.Type == "ecr" &&
+		c.RegistryAuthentication.ECRAuthData != nil &&
+		c.RegistryAuthentication.ECRAuthData.UseExecutionRole
+}
+
+// ShouldWaitForExecutionCredentials checks whether this container needs to wait for
+// credentials to pull from ECR
+func (c *Container) ShouldWaitForExecutionCredentials() bool {
```

Comment:
It seems like this method is not used anymore. Can you please get rid of this if that's the case?